q2_fondue/entrezpy_clients/_pipelines.py

```diff
@@ -27,45 +35,86 @@ def _get_run_ids(
         db = 'sra'
         elink = False
 
-    econduit = ec.Conduit(email=email, threads=n_jobs)
+    # find UIDS based on a query;
+    # instead of saving the result on the history server
+    # we will store all the UIDs recovered based on the
+    # search query and use those in the mini-pipeline below;
+    # this way we are not limited by ELink only accepting up to
+    # who knows how many IDs and erroring out if we provide too
+    # many (which could be the case e.g.: when we ask for more
+    # than 10000 BioProject IDs or the text query returns more
+    # than 10000 IDs presumably)
+    esearcher = searcher.Esearcher(
+        'esearcher', email, apikey=None,
+        apikey_var=None, threads=n_jobs, qid=None)
+    esearch_response = esearcher.inquire(
+        {
+            'db': db, 'term': term,
+            'usehistory': False, 'rettype': 'json'
+        },
+        analyzer=ESearchAnalyzer(ids))
+
+    # use the UIDs to link to other DBs and fetch related records;
+    # we won't be using multi-threading here as this shouldn't take
+    # long (we're only fetching IDs) and we don't want those dead
+    # threads afterwards
+    econduit = ec.Conduit(email=email, threads=0)
     set_up_entrezpy_logging(econduit, log_level)
-    samp_ids_pipeline = econduit.new_pipeline()
-
-    # search for IDs
-    es = samp_ids_pipeline.add_search(
-        {'db': db, 'term': " OR ".join(ids)},
-        analyzer=ESearchAnalyzer(ids)
-    )
-    if elink:
-        # given bioproject, find linked SRA runs
-        el = samp_ids_pipeline.add_link(
-            {'db': 'sra'},
-            analyzer=ElinkAnalyzer(), dependency=es
+    run_ids_pipeline = econduit.new_pipeline()
+
+    for _ids in _chunker(esearch_response.result.uids, BATCH_SIZE):
+        if elink:
+            el = run_ids_pipeline.add_link(
+                {
+                    'db': 'sra', 'dbfrom': db,
+                    'id': _ids, 'link': False
+                },
+                analyzer=ElinkAnalyzer(),
+            )
+        else:
+            el = None
+
+        # given SRA run IDs, fetch all metadata
+        efetch_params = {
+            'rettype': 'docsum', 'retmode': 'xml',
+            'reqsize': BATCH_SIZE, 'retmax': len(_ids)
+        }
+        if not elink:
+            # we need to specify these manually as in this scenario
+            # EFetch is not linked to anything
+            efetch_params.update({'id': _ids, 'db': db})
+
+        run_ids_pipeline.add_fetch(
+            efetch_params,
+            analyzer=EFetchAnalyzer(log_level), dependency=el
         )
-    else:
-        el = es
 
-    # given SRA run IDs, fetch all metadata
-    samp_ids_pipeline.add_fetch(
-        {'rettype': 'docsum', 'retmode': 'xml'},
-        analyzer=EFetchAnalyzer(log_level), dependency=el
-    )
+    econduit.run(run_ids_pipeline)
 
-    a = econduit.run(samp_ids_pipeline)
+    # recover metadata from all instances of EFetchAnalyzer
+    all_meta = []
```

Comment:
here we are recovering run IDs, right? So how about adjusting the variable name `all_meta` to `all_runids` and the comment respectively?